spec/01-unit/31-simdjson/01-cjson_compatibility_spec.lua

```diff
@@ -0,0 +1,284 @@
+        "removePageCache": "/content/admin/remove?cache=pages&id=",
+        "removeTemplateCache": "/content/admin/remove?cache=templates&id=",
+        "fileTransferFolder": "/usr/local/tomcat/webapps/content/fileTransferFolder",
+        "lookInContext": 1,
+        "adminGroupID": 4,
+        "betaServer": true}}],
+  "servlet-mapping": {
+    "cofaxCDS": "/",
+    "cofaxEmail": "/cofaxutil/aemail/*",
+    "cofaxAdmin": "/admin/*",
+    "fileServlet": "/static/*",
+    "cofaxTools": "/tools/*"},
+
+  "taglib": {
+    "taglib-uri": "cofax.tld",
+    "taglib-location": "/WEB-INF/tlds/cofax.tld"}}}
+]],
+
+[[
+{"menu": {
+    "header": "SVG Viewer",
+    "items": [
+        {"id": "Open"},
+        {"id": "OpenNew", "label": "Open New"},
+        null,
+        {"id": "ZoomIn", "label": "Zoom In"},
+        {"id": "ZoomOut", "label": "Zoom Out"},
+        {"id": "OriginalView", "label": "Original View"},
+        null,
+        {"id": "Quality"},
+        {"id": "Pause"},
+        {"id": "Mute"},
+        null,
+        {"id": "Find", "label": "Find..."},
+        {"id": "FindAgain", "label": "Find Again"},
+        {"id": "Copy"},
+        {"id": "CopyAgain", "label": "Copy Again"},
+        {"id": "CopySVG", "label": "Copy SVG"},
+        {"id": "ViewSVG", "label": "View SVG"},
+        {"id": "ViewSource", "label": "View Source"},
+        {"id": "SaveAs", "label": "Save As"},
+        null,
+        {"id": "Help"},
+        {"id": "About", "label": "About Adobe CVG Viewer..."}
+    ]
+}}
+]],
+
+[[
+[ 0.110001,
+  0.12345678910111,
+  0.412454033640,
+  2.6651441426902,
+  2.718281828459,
+  3.1415926535898,
+  2.1406926327793
+]
+]],
+
+[[
+{
+   "Image": {
+       "Width":  800,
+       "Height": 600,
+       "Title":  "View from 15th Floor",
+       "Thumbnail": {
+           "Url":    "http://www.example.com/image/481989943",
+           "Height": 125,
+           "Width":  "100"
+       },
+       "IDs": [116, 943, 234, 38793]
+     }
+}
+]],
+
+[[
+[
+   {
+      "precision": "zip",
+      "Latitude":  37.7668,
+      "Longitude": -122.3959,
+      "Address":   "",
+      "City":      "SAN FRANCISCO",
+      "State":     "CA",
+      "Zip":       "94107",
+      "Country":   "US"
+   },
+   {
+      "precision": "zip",
+      "Latitude":  37.371991,
+      "Longitude": -122.026020,
+      "Address":   "",
+      "City":      "SUNNYVALE",
+      "State":     "CA",
+      "Zip":       "94085",
+      "Country":   "US"
+   }
+]
+]],
+  }
+
+  it("runs with unified interface", function ()
+
+    local parser = simdjson.new()
+    assert(parser)
+
+    for _, str in ipairs(strs) do
+      local obj1 = parser:decode(str)
+      local obj2 = cjson.decode(parser:encode(obj1))
+      assert.same(deep_sort(obj1), deep_sort(obj2))
+    end
+
+    parser:destroy()
+  end)
+
+  it("runs with separated interface", function ()
+
+    local dec = require("resty.simdjson.decoder").new()
+    local enc = require("resty.simdjson.encoder").new()
```

Comment:
What if we add a test on the yieldable `new(true)`?

Reply:
See: https://github.com/Kong/lua-resty-simdjson/blob/main/t/04-yield.t

Reply:
How about, at Kong side, add a `new(true)` test.

Reply:
It is possible but not urgent, we can add tests later.